guava/src/com/google/common/math/DoubleMath.java

```diff
@@ -66,14 +66,14 @@ static double roundIntermediate(double x, RoundingMode mode) {
         if (x >= 0.0 || isMathematicalInteger(x)) {
           return x;
         } else {
-          return x - 1.0;
+          return (long)x - 1.0;
```

Comment:
valid point --- should check that long boundaries are not breached.

weird that there's no test coverage for this --- cause i have run all DoubleMath tests.